gateway/ln-gateway/src/db.rs

```diff
@@ -21,6 +24,88 @@ use crate::rpc::rpc_server::hash_password;
 
 pub const GATEWAYD_DATABASE_VERSION: DatabaseVersion = DatabaseVersion(1);
 
+pub trait GatewayDatabaseExt {
+    async fn save_federation_config(&self, config: FederationConfig) -> anyhow::Result<()>;
+    async fn load_federation_configs(&self) -> Vec<(FederationId, FederationConfig)>;
+    async fn get_gateway_keypair(&self) -> secp256k1::KeyPair;
+    async fn save_lnv2_incoming_contract(
+        &self,
+        payload: CreateBolt11InvoicePayload,
+    ) -> anyhow::Result<()>;
+    async fn load_lnv2_incoming_contract(
+        &self,
+        payment_hash: [u8; 32],
+    ) -> anyhow::Result<RegisteredIncomingContract>;
+}
+
+impl GatewayDatabaseExt for Database {
```

Comment:
I wonder if this wouldn't be better implemented on `DatabaseTransaction`, so that the caller is free to commit the way it wants.

Reply:
+1 to this suggestion